extensions/amp-story/1.0/variable-service.js

```diff
@@ -75,43 +75,60 @@ export class AmpStoryVariableService {
 
   /** @private */
   initializeListeners_() {
-    this.storeService_.subscribe(StateProperty.PAGE_IDS, (pageIds) => {
-      this.variables_[AnalyticsVariable.STORY_PAGE_COUNT] = pageIds.length;
-    });
+    this.storeService_.subscribe(StateProperty.PAGE_IDS, () =>
+      this.updatePageCountVariables_()
+    );
 
     this.storeService_.subscribe(
       StateProperty.CURRENT_PAGE_ID,
-      (pageId) => {
-        if (!pageId) {
-          return;
-        }
-
-        this.variables_[AnalyticsVariable.STORY_PREVIOUS_PAGE_ID] =
-          this.variables_[AnalyticsVariable.STORY_PAGE_ID];
-
-        this.variables_[AnalyticsVariable.STORY_PAGE_ID] = pageId;
-
-        const pageIndex = /** @type {number} */ (
-          this.storeService_.get(StateProperty.CURRENT_PAGE_INDEX)
-        );
-        this.variables_[AnalyticsVariable.STORY_PAGE_INDEX] = pageIndex;
-
-        const numberOfPages = this.storeService_.get(
-          StateProperty.PAGE_IDS
-        ).length;
-        if (numberOfPages > 0) {
-          if (numberOfPages === 1) {
-            this.variables_[AnalyticsVariable.STORY_PROGRESS] = 0;
-          } else {
-            this.variables_[AnalyticsVariable.STORY_PROGRESS] =
-              pageIndex / (numberOfPages - 1);
-          }
-        }
-      },
+      () => this.updatePageCountVariables_(),
       true /* callToInitialize */
     );
   }
 
+  /** @private */
+  updatePageCountVariables_() {
+    // For analytics purposes: story page count excludes ads.
+    const pageIds = /** @type {Array<string>} */ (
+      this.storeService_.get(StateProperty.PAGE_IDS)
+    );
+
+    this.variables_[AnalyticsVariable.STORY_PAGE_COUNT] = pageIds.filter(
+      (id) => !id.startsWith('i-amphtml-ad-')
+    ).length;
+
+    const pageId = /** @type {string} */ (
+      this.storeService_.get(StateProperty.CURRENT_PAGE_ID)
+    );
+    if (!pageId) {
```

Comment:
Would there be any scenario where you add an ad before the current page (because if that's the case, you need to run all the code below)?

Otherwise, `if (!pageId || this.variables_[AnalyticsVariable.STORY_PAGE_ID] === pageId)` will make it so that you don't run all the code below if the `PAGE_IDS` gets updated.

Reply:
Yes, when you start from the middle and then navigate left*7, you'll see ads.